final-project/static/scripts.js

```diff
@@ -220,207 +233,386 @@ function displayProduct(jsonOfProductInfo) {
+    prodPageElements.set('characteristics', prodPageCharacteristics);
+    prodPageElements.set('references', referenceList);
+
+
+    // Create an input element for each form.
+    prodPageElements.forEach(function(value, key) {
+
+      // Create a form element.
+      var formElement = document.createElement('form');
+      // Name the form.
+      formElement.setAttribute('name', key + '-form');
+      // Set the method to POST.
+      formElement.setAttribute('method', 'POST');
+      // POST to the url for editing the product.
+      formElement.setAttribute('action', Flask.url_for('render_product_page'));
+
+      // On the references form, loop through all the reference items.
+      if (key === 'references') {
+
+        // TODO: Set the reference id value on this UI.
+        // When the user requests reference item edits
+        // we will need to find the correct reference to update
+        // in the database (i.e. search by id).
+
+        // Loop through references.
+        for(var i = 0; i < value.length; i++) {
+
+          // Create a new form element.
+          var formElement = document.createElement('form');
+          // Name the form.
+          formElement.setAttribute('name', 'reference-form-' + i);
+          // Set the method to POST.
+          formElement.setAttribute('method', 'POST');
+          // POST to the url for editing the product.
+          formElement.setAttribute('action', Flask.url_for('render_product_page'));
+          // TODO: See if this form action can be removed
+          // based on how we submit the XHR.
+
+
+          // Create an input for the title of this reference.
+          var titleInput = document.createElement('input');
+          // Name the input.
+          titleInput.setAttribute('name', 'reference-title-input-' + i);
+          // Add placeholder text with current title.
+          titleInput.setAttribute('placeholder', value[i].firstChild.innerText);
+
+          // Label the input field with an instructive title.
+          // Create a label element.
+          var labelTitle = document.createElement('label');
+          // Link the label element to the appropriate input field.
+          labelTitle.setAttribute('for', 'reference-title-input-' + i);
+          // Set the label to be an informative one.
+          labelTitle.textContent = 'Title';
+
+          // Create an input for the link.
+          var linkInput = document.createElement('input');
+          // Name the input.
+          linkInput.setAttribute('name', 'reference-link-input-' + i);
+          // Add placeholder text with current title.
+          linkInput.setAttribute('placeholder', value[i].firstChild.href);
+          // Label the input field with an instructive title.
+          linkInput.setAttribute('label', 'Link');
+
+          // Label the input field with an instructive title.
+          // Create a label element.
+          var labelLink = document.createElement('label');
```

Comment:
Why can't this be done server-side? When using server-side templating, it's better to move away from a single-page-application and populate the logic on the server. The fact that you are doing it here makes management of templates a lot trickier.